File: 3498-reverse-degree-of-a-string/3498-reverse-degree-of-a-string.js
/**
 * @param {string} s
 * @return {number}
 */
var reverseDegree = function (s) {
    let sum = 0;
    let product = 1;
    for (let i = 0; i < s.length; i++) {
        let index = i + 1;
        let original = s[i].charCodeAt() - 'a'.charCodeAt(0);
        let reversed = 26 - original;
        product = index * reversed;
        sum += product;
    }

    return sum;
};

File: 3498-reverse-degree-of-a-string/3498-reverse-degree-of-a-string.java
class Solution {
    public int reverseDegree(String s) {
        int sum = 0;
        int product = 1;

        for (int i = 0; i < s.length(); i++) {
            int index = i + 1;
            int original = s.charAt(i) - 'a';
            int reversed = 26 - original;
            product = index * reversed;
            sum += product;
        }
        return sum;
    }
}